lib/job-simulation.rb

```diff
@@ -2,19 +2,47 @@
 require './Queue.rb'
 
 class JobSimulation
-  attr_reader :workers, :waiting, :roll
+  #attr_reader :workers, :waiting, :roll
 
-  def initialize (jobs_available, job_seekers)
+  def initialize (job_seekers)
+    jobs_available = 6
+    jobs_waiting = job_seekers - jobs_available
+
+    @workers = Stack.new
+    jobs_available.times do |i|
+      @workers.push("Worker ##{ i + 1 }")
+    end
+
+    @waiting = Queue.new
+    jobs_waiting.times do |i|
+      @waiting.enqueue("Worker ##{ i + jobs_available + 1 }")
+    end
 
   end
 
   def cycle
+    roll = rand(1..6)
+    puts "Cycling #{ roll } employees"
+    roll.times do |i|
+      @waiting.enqueue(@workers.pop)
+    end
+
+    roll.times do |i|
+      @workers.push(@waiting.dequeue)
+    end
+  end
 
+  def workers
+    @workers.to_s
+  end
+
+  def waiting
+    @waiting.to_s
   end
```

Comment:
`def workers` and `def waiting` are not needed because of the `attr_reader` at the top of the class. These should be removed.